extensions/resteasy-classic/rest-client-config/runtime/src/main/java/io/quarkus/restclient/config/RestClientsConfig.java

```diff
@@ -327,19 +328,27 @@ public RestClientConfig getClientConfig(String configKey) {
         if (configKey == null) {
             return RestClientConfig.EMPTY;
         }
-        return configs.computeIfAbsent(configKey, RestClientConfig::load);
+        CompletableFuture<RestClientConfig> futureConfig = configs.computeIfAbsent(configKey, name -> new CompletableFuture<>());
+        if (!futureConfig.isDone()) {
```

Comment:
This won't work as expected.
You need to make sure that there is competition here and load is performed just once; the only way to to make sure they who performed the call has been the race to set the completable future as first, while here you risk 2 threads to try complete whatever they found or compute in the CHM

Reply:
I agree that it is possible that `load` could be performed more than once by concurrent threads (which is indeed suboptimal), but in this case the `CompletableFuture` will act as a sort of `AtomicReference` and only the first thread completing the config loading will win and set that value inside the `CompletableFuture`.

> the only way to to make sure they who performed the call has been the race to set the completable future as first, while here you risk 2 threads to try complete whatever they found or compute in the CHM

Sorry, but I'm not understanding what you're suggesting here. Can you please clarify, possibly with a practical example, how I could avoid that 2 threads could concurrently call the `load` method?

Reply:
The thing is that we shouldn't introduce unwanted behaviour from what it was before, or we risk to introduce bugs because, being concurrent is not always due that is well tested/have coverage.

My suggestion is to preserve the original load behaviour and yo do it you need some atomic way to set the completable future and being aware that the result of attempting to create it has had success, instead of competing to complete it.

There are different ways to do it but I have no computer on my hand now to give you a detailed example.
Before CHM::computeIfAbsent was thing it was used a fast path get and a slow path putIfAbsent which create a completable future upfront: iirc part of the putIfAbsent contract is the ability to verify if it has landed (or there much be another method which allow to do it, for sure) and who won the race to set it, will perform the load and complete it, who loose will just perform a waiting get. I hope it helps to clarify

Reply:
@franz1981 thanks for clarification, I wasn't thinking to use the `putIfAbsent` to make sure that the load is performed only once. I modified this pull request according to your suggestion, please review again.

@radcortez feedback is welcome.

Reply:
All good!